### ISSUE TITLE
Connect to HTTP trackers fails if bind-address-ipv6 is set AND bind-address-ipv4 is not

### ISSUE THREAD

**Issue by s3phy** [transmission/transmission#88]
Hi,

I recently realized my Transmission wasn't able to talk to HTTP trackers anymore. Today I finally dug to the source of the problem, and found it to be linked to the fact that I had configured "bind-address-ipv6" in my settings.json a few months ago. IPv6 works perfectly fine with the IP I want transmission to bind to configured that way.

With an IPv6 set in bind-address-ipv6, connecting to HTTP trackers fail, with the following message using TR_CURL_VERBOSE=1 :
```
[2016-11-18 01:16:36.926] <torrent name> Announcing to tracker (announcer.c:1552)
* Hostname was NOT found in DNS cache
*   Trying <tracker's ipv4 addr>...
* Hostname was NOT found in DNS cache
* Name '<ipv6 addr configured in bind-address-ipv6>' family 2 resolved to '<ipv6 addr configured in bind-address-ipv6>' family 10
* bind failed with errno 22: Invalid argument
* Closing connection 0
[2016-11-18 01:16:37.926] <torrent name> Could not connect to tracker (announcer.c:996)
[2016-11-18 01:16:37.926] <torrent name> Retrying announce in 20 seconds. (announcer.c:1005)
[2016-11-18 01:16:57.926] <torrent name> Announcing to tracker (announcer.c:1552)
* Hostname was found in DNS cache
*   Trying <tracker's ipv4 addr>...
* Hostname was found in DNS cache
* Name '<ipv6 addr configured in bind-address-ipv6>' family 2 resolved to '<ipv6 addr configured in bind-address-ipv6>' family 10
* bind failed with errno 22: Invalid argument
* Closing connection 1
```

With the default "::" in bind-address-ipv6, connecting to HTTP trackers works fine, but of course transmission doesn't bind itself to the IPv6 address I want it to…

transmission 2.92, compiled myself, using transmission-daemon in command line on Debian 8.6

Ticket #5738 on TRAC (is TRAC still used?) seems related to this bug: https://trac.transmissionbt.com/ticket/5738 at the difference that unlike this other report, I have a working IPv6 connectivity.

Best regards.

**Comment by s3phy** [comment]
I dug even more to try to understand what's happening. I think the problem is linked to a behavior of curl: whatever is passed to curl as --interface, it tries to use it to open the connection to the tracker.

The tracker in my case being only in IPv4, curl tries to open a connection from IPv6 to IPv4 and of course it fails.


Digging even further (and trying to understand how transmission is using curl, instead of focusing on how curl works) I found a work-around: looking at libtransmission/web.c, in "static CURL *
createEasy", around lines 208 ~ 211 (in Transmission 2.92), I see that CURLOPT_INTERFACE is only filled if the bind interface configured in transmission is different from the default value, AND that it is checked for IPv4 before IPv6, with an elseif, in fact never setting CURLOPT_INTERFACE with an IPv6 as long as bind-address-ipv4 is set.

I had left bind-address-ipv4 to its default value, because I didn't need/want to set a specific value. I have filled it with my box's IPv4 address, address which is now given to curl to use as CURLOPT_INTERFACE, not giving it the IPv6 I set in bind-address-ipv6, and everything is happy.

I think this should be at least documented (bind-address-ipv4 needing to be set if bind-address-ipv6 is set if we want to be able to connect to IPv4-only HTTP trackers).

**Comment by mshick** [comment]
Thank you so much for posting this workaround. I've been trying to disable ipv6 for Transmission only, due to some tracker problems, and was incredibly confused by the failure to find an open port with a link-local ipv6.

After reading your issue, I now understand what was happening, and was able to workaround my problems fully.

Would love for this to be documented until a true fix is put in place.

**Comment by albel727** [comment]
Great job figuring this bug out!

I was hit by this issue, having set my `bind-address-ipv6` to `::1` a long time ago, in order to disable ipv6, as was advised in some relevant ticket. 

I hadn't noticed that this actually broke http trackers until just now. And then I wouldn't have realized the connection with the bind setting, if I hadn't actually had two transmission-daemons running, differing only in `bind-address-ipv6` setting, and the one having the default `bind-address-ipv6 = "::"` correctly working with http trackers.

Thanks for confirming my suspicions, and indeed, setting `bind-address-ipv6 = "::"` resolves my tracker problems.

**Comment by ieee8023** [comment]
This is still happening as of "Transmission 3.00 (180586d1ba)"
HTTP and HTTPS requests for both trackers and to add torrents via URLs return "http error 0: No Response"

When adding a torrent via URL over HTTP or HTTPS I would get: "gotMetadataFromURL: http error 0: No Response"

**Comment by tearfur** [comment]
Hopefully fixed by #5296
